### ISSUE TITLE
Network start/end event emitted more than once

### ISSUE THREAD

**Issue by hayesmg** [noflo/noflo#221]
The `start` and `end` events for a network are emitted more than once.  This appears to be caused by the connect/send/disconnect of initials and now defaults.  The `increaseConnections` and `decreaseConnections` functions work solely off of `@connectionCount`.  In the case of `increaseConnections`, a `and not @startEmitted` statement would suffice; `@startEmitted` being a boolean flag.  `decreaseConnections` will be a bit more tricky.

Example code:

``` js
function createComp() {
    var o = new noflo.Component();
    o.inPorts.add('in', {datatype: 'all', required: true }, function(e, data) {
        if (e === 'data')
            console.log(data);
    });
    return o;
}

var fbp = '\'Input1\' -> IN Output1(Output)\n\'Input2\' -> IN Output2(Output)';

noflo.graph.loadFBP(fbp, function(g) {
    noflo.createNetwork(g, function(n) {

        n.loader.components.Output = createComp;

        n.on('start', function() {
            console.log('network started!');
        });
        n.on('end', function() {
            console.log('network end!');
        });

        n.connect(n.start.bind(n));
    }, true);
})
```

Output:

```
network started!
Input1
network started!
Input2
network end!
network end!
```

Issue first identified by akonsu on SO: http://stackoverflow.com/questions/24522055/noflo-network-emits-start-and-end-events-multiple-times

**Comment by jonnor** [comment]
Is this issue still present?

**Comment by dremekie** [comment]
@jonnor 
This is is still an issue. Take the following graph.

```
{
    "processes": {
        "WriteFile": {
            "component": "WriteFile",
            "metadata": {}
        },
        "Display": {
            "component": "core/Output",
            "metadata": {}
        }
      },
    "connections": [
        {
            "data": "/Users/dwayne/Documents/test.txt",
            "tgt": {
                "process": "WriteFile",
                "port": "filename"
            }
        },
        {
            "src": {
                "process": "WriteFile",
                "port": "out"
            },
            "tgt": {
                "process": "Display",
                "port": "in"
            }
        },
        {
            "src": {
                "process": "WriteFile",
                "port": "error"
            },
            "tgt": {
                "process": "Display",
                "port": "in"
            }
        },
        {
            "data": "Some useless text",
            "tgt": {
                "process": "WriteFile",
                "port": "in"
            }
        }
    ]
}
```

And the following code snippet

```
  noflo.graph.loadJSON(nofloDefinition, function(graph) {
    noflo.createNetwork(graph, function(network) {
      network.on('start', function() {
        console.log('network started!');
      });
      network.on('end', function() {
        console.log('network end!');
      });
    });
  });
```

I see this:
```
network started!
network started!
network started!
/Users/dwayne/Documents/test.txt
network end!
network end!
network end!
```

Is it even possible to definitely know when a network is "complete"?

**Comment by bergie** [comment]
@dremekie I just tried your graph and runner code (adapting them to my local filesystem path, and error callbacks), and I get:

```
⇒  node run.js             
test.txt
network end!
``

The versions used are:

```
    "noflo": "^0.8.3",
    "noflo-core": "^0.4.0",
    "noflo-filesystem": "^1.1.1"
```

So from that, it seems this has been fixed in NoFlo 0.8
